spec/support/CurrentSpecReporter.js

```diff
@@ -1,15 +1,45 @@
 // Sets a global variable to the current test spec
 // ex: global.currentSpec.description
-
+const { performance } = require('perf_hooks');
 global.currentSpec = null;
 
+const timerMap = {};
+const duplicates = [];
+/** The min. execution time in seconds for a test to be considered slow. */
+const slowTestLimit = 2;
+
 class CurrentSpecReporter {
   specStarted(spec) {
+    if (timerMap[spec.fullName]) {
+      console.log('Duplicate spec: ' + spec.fullName);
+      duplicates.push(spec.fullName);
+    }
+    timerMap[spec.fullName] = performance.now();
     global.currentSpec = spec;
   }
-  specDone() {
+  specDone(result) {
+    if (result.status === 'excluded') {
+      delete timerMap[result.fullName];
+      return;
+    }
+    timerMap[result.fullName] = (performance.now() - timerMap[result.fullName]) / 1000;
     global.currentSpec = null;
   }
 }
+global.displaySlowTests = function() {
+  const times = Object.values(timerMap).sort((a,b) => b - a);
+  if (times.length > 0) {
+    console.log(`Slow tests with execution time >=${slowTestLimit}s:`);
+  }
+  times.forEach((time) => {
+    if (time >= slowTestLimit) {
+      console.warn(`${time.toFixed(1)}s:`, Object.keys(timerMap).find(key => timerMap[key] === time));
```

Comment:
Cut down to 1 digits, since I found there is an execution time tolerance in the ~0.05s range anyway.

Reply:
Nice! I figure he handle the big test then focus on the smaller ones. 1 second works for me

Reply:
It's actually 1 digit after the comma, so the log is `2.2s` for example.

Reply:
lI was optimizing milliseconds when I wrote it loo

Reply:
We can put ms if you believe it is useful. I just noticed that there is a variance in execution time, so beyond 0.05 seconds there doesn't seem to be a reliable stability for the metric to optimize based on it.

Reply:
I can change the code locally at anytime to fit my needs. We can keep the current implementation as is.